tests/phpunit/tests/test-static-pages.php

```diff
@@ -312,6 +312,42 @@ function test_page_for_posts() {
 		$this->assertEquals( array( get_post( $en ) ), $GLOBALS['wp_query']->posts );
 	}
 
+	function test_untranslated_page_for_posts_2() {
+		wp_delete_post( self::$posts_fr, true );
```

Comment:
I don't recall that we tested manually what happens when the page for posts translations is deleted. This might be different from what happens when it doesn't exists in the first place.

Reply:
I tested both cases manually, when I didn't translate it and when I translated and then deleted it.